code/addons/storysource/src/StoryPanel.tsx

```diff
@@ -58,14 +58,14 @@ export const StoryPanel: React.FC<StoryPanelProps> = ({ api }) => {
   // prefer to use the source from source-loader, but fallback to
   // source provided by csf-plugin for vite usage
   const source = loaderSource || docsSource || 'loading source...';
-  const currentLocation = locationsMap
-    ? locationsMap[
-        Object.keys(locationsMap).find((key: string) => {
-          const sourceLoaderId = key.split('--');
-          return story.id.endsWith(sourceLoaderId[sourceLoaderId.length - 1]);
-        })
-      ]
+  const currentLocationIndex = locationsMap
+    ? Object.keys(locationsMap).find((key: string) => {
+        const sourceLoaderId = key.split('--');
+        return story.id.endsWith(sourceLoaderId[sourceLoaderId.length - 1]);
+      })
     : undefined;
+  const currentLocation =
+    !!locationsMap && !!currentLocationIndex ? locationsMap[currentLocationIndex] : undefined;
   React.useEffect(() => {
```

Comment:
I think the `!!` is unnecessary here right? I think this is not a code style we commonly use. I would rather write it like this:
```ts
 locationsMap && currentLocationIndex ? 
```
or
```ts
 locationsMap != null && currentLocationIndex != null ? 
```

Reply:
It's a commonly-used shorthand for casting to Boolean, but I usually recommend to developers in my team to just use `Boolean(x)` instead of trying to save 5 bytes at the expense of readability.